### DISCUSSION TITLE
GPS-RTK2 lack of integrated compass

### DISCUSSION THREAD

**Discussion by willishf** [iNavFlight/inav#8947] [General]
Working with the ZED-F9P GPS from Ublox(https://www.sparkfun.com/products/15136) that has 1 cm accuracy. We have standardized on this for our robot navigation and trying to see how we can leverage on a drone platform for precise location-based imaging in agriculture using a drone. 

The ZED-F9P does not include a compass, and based on what I have read in INAV support forums the recommendation is to use low-cost GPS units that have a magnetometer on the same board but integrate with I2C.

I ordered a couple different HMC5883(magnetometer) boards for testing to see if I can get INAV to recognize. 

Brings up the larger discussion of flying without a compass(noise) and using GPS heading as a proxy for compass. I realized that this has been discussed in prior feature requests as not a great option and, of course, depends on the drone platform. 

In our particular use case, we want to automate a mission to fly over a multi-acre field and take high-resolution images that would allow the early detection of crop problems and track progress on the same plant over multiple missions. 

With 1cm GPS accuracy at the time a photo is taken doable. Does create an orientation problem, and we plan on having two GPS-RTK2 on the drone and/or fix-wing that would give precise orientation. We can calculate a very accurate heading in real-time even when not moving, and assume we can set that value via MSP. It is on our testing list, but still trying to get the base drone built and configured. 

Mainly wanted to bring up the issue we are running into that very difficult to find a newer FC with an integrated compass, and the latest GPS modules from U-BLOX with RTK2 support don't have an integrated compass. Longer term, this will be an issue for iNAV on flight mission support as a feature and why a GPS-derived virtual heading would be a good option.  

If we get the separate HMC5883 standalone boards working then a good solution moving forward. Didn't see that as a recommended option for a specific board to use in the documentation related to the compass.

**Comment by stronnag** [comment]
1. Onboard compass has demonstrated to be almost useless, as it's practically impossible to remove from interference. Explains lack of availability.
2. Use an external I2C compass. Whatever is supported for your FC.
